### PR TITLE
Expose the number of tokens owned by an ingester

### PR DESCRIPTION
**Pull request by gouthamve**
* Also make sure the percentage is exposed for ingesters
  that are not having any tokens

### CODE REVIEW

**Comment by gouthamve** [comment]
This will help with alerting for #954 If there are 0 tokens, the `ownership_percentage` series is missing.

**Comment by tomwilkie** [comment]
These metrics will be exposed by distributor & querier - would be good to have some exposed by the ingester (through lifecycle.go)

File: pkg/ring/ring.go [hunk 1]
```diff
@@ -130,11 +131,16 @@ func New(cfg Config) (*Ring, error) {
...
+		totalTokensDesc: prometheus.NewDesc(
+			"cortex_ring_tokens_total",
 			"Number of tokens in the ring",
 			nil, nil,
 		),
+		numTokensDesc: prometheus.NewDesc(
+			"cortex_ring_tokens_owned",
+			"The number of tokens in the ring owned by the ingester",
+			[]string{"ingester"}, nil,
+		),
```
Comment (tomwilkie) [inline pkg/ring/ring.go]: Can we make these be global and use promauto whilst we're here?

**Comment by csmarchbanks** [comment]
Why do we need the new count metric? Is fixing the percentage metric not enough?

File: pkg/ring/ring.go (hunk 1 above)
Comment (gouthamve) [reply pkg/ring/ring.go]: Thinking about it, it'll be bad idea to make this global. We need to just expose the active ingesters, not all the ingester throughout the lifetime of the distributor and having it the current way enables us to do it.

File: pkg/ring/lifecycler.go [hunk 2]
```diff
@@ -20,16 +21,20 @@ const (
 )
 
 var (
-	consulHeartbeats = prometheus.NewCounter(prometheus.CounterOpts{
+	consulHeartbeats = promauto.NewCounter(prometheus.CounterOpts{
 		Name: "cortex_ingester_consul_heartbeats_total",
 		Help: "The total number of heartbeats sent to consul.",
 	})
+	tokensOwned = prometheus.NewGauge(prometheus.GaugeOpts{
```
[the remark this answers is not in the document]
Comment (gouthamve) [reply pkg/ring/lifecycler.go]: Oops, yep! Fixed!

File: pkg/ring/ring.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/ring/ring.go]: Fair enough!

**Pull request merged**